### PR TITLE
Make the S2S fetch self-healing and alert on stale providers

### PR DESCRIPTION
**Pull request by seribaymadina**
The fetch read only each provider's newest completed Coval run and silently skipped it when already ingested, so a paused schedule or a stuck sim left the dashboard frozen with no signal. Each tick now scans the recent window (two fetch periods, floored at one day), ingests every clean not-yet-ingested run slotted by its create_time on the fetch grid, and never ingests runs finalized with an error_status. A provider whose newest usable run is older than the fetch period (plus an optional grace, default zero) is reported FAILED, which flows into the existing RUN_FAILED log alert — a missed sim round at the daily cadence gets flagged on the next fetch instead of going unnoticed. Also adds unit tests for the fetch module, which had none.

<h3>Greptile Summary</h3>

This PR makes the S2S Coval fetch scan recent runs and report stale providers. The main changes are:

- Adds a configurable stale grace period.
- Ingests recent clean runs instead of only the newest completed run.
- Slots ingested runs by Coval `create_time`.
- Skips runs finalized with an error status.
- Adds unit tests for the S2S fetch path.

<h3>Confidence Score: 4/5</h3>

This is close, but the paginated fetch path should be fixed before merging.

- The stale backfill and missing timestamp cases now fail closed.
- The runs scan still stops after the first page, so valid recent runs can be skipped.
- The new tests cover freshness behavior, but they only model single-page responses.

runner/src/coval_bench/s2s/fetch_v2v.py

<sub>Reviews (2): Last reviewed commit: ["Let staleness override backfill-only and..."](https://github.com/coval-ai/benchmarks/commit/5625f3ba7a301b198add7b84734e4aa880405bdc) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=44226651)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

File: runner/src/coval_bench/s2s/fetch_v2v.py [hunk 1]
```diff
@@ -76,50 +91,67 @@ def _dataset_sha256() -> str:
...
             "filter": f'status="COMPLETED" AND agent_id="{agent_id}"',
             "order_by": "-create_time",
-            "page_size": 1,
+            "page_size": WINDOW_PAGE_SIZE,
         },
     )
     resp.raise_for_status()
-    runs = cast("list[dict[str, Any]]", resp.json().get("runs", []))
-    return cast("str | None", runs[0]["run_id"]) if runs else None
+    raw = cast("list[dict[str, Any]]", resp.json().get("runs", []))
```
[the remark this answers is not in the document]
Comment (seribaymadina) [reply runner/src/coval_bench/s2s/fetch_v2v.py:128]: won't bite at this scale

File: runner/src/coval_bench/s2s/fetch_v2v.py [hunk 2]
```diff
@@ -76,50 +91,67 @@ def _dataset_sha256() -> str:
...
+    better to ingest with a fetch-time slot than to drop data.
     """
+    window_seconds = max(WINDOW_FLOOR_SECONDS, 2 * period_seconds)
     resp = await client.get(
         "/runs",
         params={
             "filter": f'status="COMPLETED" AND agent_id="{agent_id}"',
             "order_by": "-create_time",
-            "page_size": 1,
+            "page_size": WINDOW_PAGE_SIZE,
```
[the remark this answers is not in the document]
Comment (seribaymadina) [reply runner/src/coval_bench/s2s/fetch_v2v.py:124]: agent won't have more than 10 completed runs in our case

**Review by cooperr1 — APPROVED** [review]
